app/controllers/posts_controller.rb

```diff
@@ -30,6 +32,17 @@ def index
     @posts = Post.where(author_id: @user.id).includes(:comments, :user).paginate(page: params[:page], per_page: 10)
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    authorize! :destroy, @post
+    if @post.destroy
+      flash.now[:success] = 'Post was successfully destroyed!'
+    else
+      flash[:error] = 'Post couldnt be destroyed!'
+    end
+    redirect_to user_posts_path
+  end
+
```

Comment:
- [ ] As requested by the previous reviewer kindly ensure deleting of a post by the user who created it is possible

Videocast
[cinnamon-2023-12-23T153822+0200.webm](https://github.com/maths29/blog_app/assets/81932478/27bad636-1184-46a6-aee5-a01eb1519e49)

- [ ] Also creating a comment now is not possible, kindly fix the error that occurs each time a user  tries to create a comment for a post
[cinnamon-2023-12-23T154014+0200.webm](https://github.com/maths29/blog_app/assets/81932478/e57bded8-1a96-496c-a15c-2c1da5cad832)